akeneo-design-system/src/components/Input/TagInput/TagInput.tsx

```diff
@@ -0,0 +1,223 @@
+import React, {useState, useRef, ChangeEvent, FC, RefObject, KeyboardEvent, useCallback} from 'react';
+import styled from 'styled-components';
+import {AkeneoThemedProps, getColor} from '../../../theme';
+import {CloseIcon} from '../../../icons';
+import {Key} from '../../../shared';
+
+type TagInputProps = {
+  /**
+   * Specifies if the component will accept duplicated tags or not
+   */
+  allowDuplicates: boolean;
+
+  /**
+   * Tags to display
+   */
+  tags: string[];
+
+  /**
+   * Handle called when tags are updated
+   */
+  setTags: (tags: string[]) => void;
+
+  /**
+   * Placeholder displayed where there is no tag
+   */
+  placeholder?: string;
+
+  /**
+   * Defines if the input is valid on not.
+   */
+  isInvalid?: boolean;
```

Comment:
```suggestion
  invalid?: boolean;
```

On text input, the validity is named `invalid`. either way, I'm ok but it could be nice to have some sort of convention IMHO